src/L1/rollup/ScrollChain.sol

```diff
@@ -315,78 +347,204 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
     }
 
     /// @inheritdoc IScrollChain
-    function commitBatch(
+    ///
+    /// @dev This function will revert unless all V0/V1/V2 batches are finalized. This is because we start to
+    /// pop L1 messages in `commitBatchWithBlobProof` but not in `commitBatch`. We also introduce `finalizedQueueIndex`
+    /// in `L1MessageQueue`. If one of V0/V1/V2 batches not finalized, `L1MessageQueue.pendingQueueIndex` will not
+    /// match `parentBatchHeader.totalL1MessagePopped` and thus revert.
+    ///
+    /// @dev `_skippedL1MessageBitmap` is no longer used, will remove in next version.
+    function commitBatchWithBlobProof(
         uint8 _version,
         bytes calldata _parentBatchHeader,
         bytes[] memory _chunks,
-        bytes calldata _skippedL1MessageBitmap
-    ) external override OnlySequencer whenNotPaused {
-        (bytes32 _parentBatchHash, uint256 _batchIndex, uint256 _totalL1MessagesPoppedOverall) = _beforeCommitBatch(
-            _parentBatchHeader,
-            _chunks
-        );
+        bytes calldata, /*_skippedL1MessageBitmap*/
+        bytes calldata _blobDataProof
+    ) external override {
+        commitBatchWithBlobProof(_version, _parentBatchHeader, _chunks, _blobDataProof);
+    }
 
-        bytes32 _batchHash;
-        uint256 batchPtr;
-        bytes32 _dataHash;
-        uint256 _totalL1MessagesPoppedInBatch;
-        if (1 <= _version && _version <= 2) {
-            // versions 1 and 2 both use ChunkCodecV1 and BatchHeaderV1Codec,
-            // but they use different blob encoding and different verifiers.
-            (_dataHash, _totalL1MessagesPoppedInBatch) = _commitChunksV1(
-                _totalL1MessagesPoppedOverall,
-                _chunks,
-                _skippedL1MessageBitmap
-            );
-            assembly {
-                batchPtr := mload(0x40)
-                _totalL1MessagesPoppedOverall := add(_totalL1MessagesPoppedOverall, _totalL1MessagesPoppedInBatch)
+    /// @inheritdoc IScrollChain
+    function commitBatchWithBlobProof(
```

Comment:
Is this interface used for rollup-relayer upgrading commit interface asynchronously?